### DISCUSSION TITLE
DBCopy from opencrx 4.1 to 5.1

### DISCUSSION THREAD

**Discussion by nyemiks** [opencrx/opencrx-documentation#30] [Q&A]
I have installed the opencrx v5.1 db objects to a new database. I have some custom tables that were created from custom models in the opencrx 4.1 database. Once, I have successfully migrated my custom opencrx 4.1 project to custom opencrx 5.1 project. I intend to deploy the custom tables as well.

My question is, can I use DbCopy to copy all my data from opencrx4.1 to opencrx5.1 database ? Are there likely to be challanges due to the version difference ?

Will it copy data from my custom tables as well ?

**Comment by wfro64** [comment]
DbSchemaUtils and DbCopy take into account all custom extensions if they have been added to `src/main/resources/META-INF/dbschema-add.sql`. E.g. see https://github.com/opencrx/opencrx-custom/tree/master/sample/src/main/resources/META-INF.

Note: DbSchemaUtils.getSchemaConnection() reads the core schema into an in-memory HSQLDB and applies the custom extensions configured in dbschema-add.sql. This reference schema is then used by DbSchemaUtils, DbCopy, ...
